node_test/helpers_test.go

```diff
@@ -185,20 +185,39 @@ func setupLedgerChannel(t *testing.T, alpha node.Node, beta node.Node, asset com
 		t.Fatal(err)
 	}
 
+	if sim != nil {
+		// This delay + block commit ensures chainservice event subs trigger and process events
+		time.Sleep(100 * time.Millisecond)
+		sim.Commit()
+	}
+
+	t.Log("Waiting for direct-fund objective to complete...")
+
 	<-alpha.ObjectiveCompleteChan(response.Id)
 	<-beta.ObjectiveCompleteChan(response.Id)
 
+	t.Log("Completed direct-fund objective")
+
 	return response.ChannelId
 }
 
-func closeLedgerChannel(t *testing.T, alpha node.Node, beta node.Node, channelId types.Destination) {
+func closeLedgerChannel(t *testing.T, sim chainservice.SimulatedChain, alpha node.Node, beta node.Node, channelId types.Destination) {
 	response, err := alpha.CloseLedgerChannel(channelId)
 	if err != nil {
 		t.Fatal(err)
 	}
 
+	if sim != nil {
+		time.Sleep(100 * time.Millisecond)
```

Comment:
What is the reason for this delay?

Reply:
This does feel prone to future errors because anytime we send a transaction when using the Simulated Backend we may need to add this delay. Perhaps it would be better to just add the delay inside the [`SendTransaction`](https://github.com/statechannels/go-nitro/blob/newBlock-listener/node/engine/chainservice/simulated_backend_chainservice.go#L86) method directly so we don't have to do it externally?

Reply:
Made the change I mentioned above in this commit: https://github.com/statechannels/go-nitro/pull/1517/commits/a1b1afd1c3cfbd5ab7ea6aa04c2790aff0514d5c